java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -379,37 +362,26 @@ private API getXPAPI(
       MemberSelectTree mst = ((MemberSelectTree) mit.getMethodSelect());
       // when returnType is specified, check if mst's return type matches it
       // if it's not a match, skip to next method property map
-      String returnType = getValueStringFromMap(currentMethodProperties, RETURN_TYPE_STRING);
-      if (returnType != null) {
+      if (methodRecord.getReturnType().isPresent()) {
         String mReturn = ASTHelpers.getReturnType(mst).toString();
-        if (!returnType.equals(mReturn)) {
+        if (!methodRecord.getReturnType().get().equals(mReturn)) {
```

Comment:
Store `methodRecord.getReturnType()` to avoid verbosity?

Reply:
See above.